src/MergeContext.js

```diff
@@ -1295,6 +1314,9 @@ class PullRequest {
             else
                 this._removePositiveStagingLabels();
 
+            // some of these labels may be set already in the exception
+            this._addPrStateLabels();
```

Comment:
AFAICT, to do this addition here, you have to move `if (!this._labels)` from below to here first. The "report this..." comment attached to that `if` statement should stay where it is now, of course.